client/express-checkout/utils/normalize.js

```diff
@@ -288,6 +288,25 @@ const getAdditionalFieldsDataFromStore = () => {
 	return store.getAdditionalFields() || {};
 };
 
+/**
+ * Get extension data from the checkout store.
+ *
+ * @return {Object} The extension data.
+ */
+const getExtensionDataFromStore = () => {
+	const checkoutStore = window.wc?.wcBlocksData?.checkoutStore;
+	if ( ! checkoutStore ) {
+		return {};
+	}
+
+	const store = select( checkoutStore );
+	if ( ! store ) {
+		return {};
+	}
+
+	return store.getExtensionData() || {};
```

Comment:
Should we be a bit more defensive to make sure that `getExtensionData()` is defined?
```suggestion
	if ( ! store || ! store?.getExtensionData ) {
		return {};
	}

	return store.getExtensionData() || {};
```